### PR TITLE
T37615 kernelci_api: add 'Database.count_nodes()' method

### PR DESCRIPTION
**Pull request by JenySadadia**
Depends on https://github.com/kernelci/kernelci-api/pull/179

Implement database method to get the count of all nodes matching attributes from API.

Signed-off-by: Jeny Sadadia <jeny.sadadia@collabora.com>

### CODE REVIEW

File: kernelci/db/kernelci_api.py [hunk 1]
```diff
@@ -88,6 +88,10 @@ def get_node(self, node_id):
         resp = self._get('/'.join(['node', node_id]))
         return json.loads(resp.text)
 
+    def get_nodes_count(self, attributes: dict = None):
```
[the remark this answers is not in the document]
Comment (JenySadadia) [reply kernelci/db/kernelci_api.py]: Seems OK too.

File: kernelci/db/kernelci_api.py [hunk 2]
```diff
@@ -72,6 +72,11 @@ def unsubscribe(self, sub_id):
             self._filters.pop(sub_id)
         self._post(f'unsubscribe/{sub_id}')
 
+    def count_nodes(self, attributes: dict = None):
+        """Get the count of all nodes matching attributes"""
+        resp = self._get('count', params=attributes)
+        return resp.json()
+
```
Comment (gctucker) [inline kernelci/db/kernelci_api.py:79]: Maybe this should have been put after `get_nodes()` to keep the pub/sub methods grouped together, then the `nodes` ones together.  Not a big issue though.

Comment (JenySadadia) [reply kernelci/db/kernelci_api.py:79]: I wanted to maintain the alphabetical order for methods related to nodes. That's why I put it first after pubsub methods.

Comment (gctucker) [reply kernelci/db/kernelci_api.py:79]: But, `get_event()` is not related to nodes.  So it should at least be put after that.

Comment (JenySadadia) [reply kernelci/db/kernelci_api.py:79]: Oh, yes. I will update it.

**Review by gctucker — APPROVED** [review]

**Pull request merged**